ikura/shim/src/cli.rs

```diff
@@ -120,19 +108,67 @@ pub enum Commands {
 pub mod serve {
     //! CLI definition for the `serve` subcommand.
 
-    use super::{DockParams, IkuraRpcParams, KeyManagementParams};
-    use clap::Args;
+    use super::{DockParams, IkuraRpcParams, KeyManagementParams, ENV_IKURA_NAMESPACE};
+    use clap::{Args, Subcommand};
 
     #[derive(Debug, Args)]
     pub struct Params {
-        #[clap(flatten)]
-        pub rpc: IkuraRpcParams,
+        #[command(subcommand)]
+        pub dock: Dock,
+    }
+
+    #[derive(Subcommand, Debug)]
+    pub enum Dock {
+        /// Serve requests of the Sovereign SDK rollups.
+        Sov(sov::Params),
+        /// Serve requests of the Rollkit SDK rollups.
+        Rollkit(rollkit::Params),
+    }
 
-        #[clap(flatten)]
-        pub dock: DockParams,
+    pub mod sov {
+        //! CLI definition for the `serve sov` subcommand.
 
-        #[clap(flatten)]
-        pub key_management: KeyManagementParams,
+        use super::{DockParams, IkuraRpcParams, KeyManagementParams};
+        use clap::Args;
+
+        #[derive(Debug, Args)]
+        pub struct Params {
+            #[clap(flatten)]
+            pub rpc: IkuraRpcParams,
+
+            #[clap(flatten)]
+            pub dock: DockParams,
+
+            #[clap(flatten)]
+            pub key_management: KeyManagementParams,
+        }
+    }
+
+    pub mod rollkit {
+        //! CLI definition for the `serve rollkit` subcommand.
+
+        use super::{DockParams, IkuraRpcParams, KeyManagementParams, ENV_IKURA_NAMESPACE};
+        use clap::Args;
+
+        #[derive(Debug, Args)]
+        pub struct Params {
+            #[clap(flatten)]
+            pub rpc: IkuraRpcParams,
+
+            #[clap(flatten)]
+            pub dock: DockParams,
+
+            #[clap(flatten)]
+            pub key_management: KeyManagementParams,
+
+            /// The namespace to submit the blobs into.
+            ///
+            /// The namespace can be specified either as a 16-byte vector, or as an unsigned 128-bit
+            /// big-endian integer. To distinguish between the two, the byte vector must be prefixed
+            /// with `0x`.
+            #[clap(long, short, env = ENV_IKURA_NAMESPACE)]
+            pub namespace: Option<String>,
```

Comment:
You mentioned in the PR description that rollkit doesn't submit blobs with a namespace. Also, when this is `None` we get a CLI warning.

Bu the `SubmitBlob` gRPC contains a `namespace`.

What I am understanding so far is that:
  1. With any rollkit chain today, the namespace is omitted in the request and therefore must be configured in the shim.
  2. Rollkit could be altered in the future to support sending namespaces without requiring any changes to the shim

correct?

Reply:
Kind of.

I am not sure what is the reason of this interface and what is the ultimate game plan: maybe the namespace being phased out? Or maybe it got a hit during migration to go-da. Either way, right now  it doesn't seem that Rollkit takes the namespace ID.

I just went to check to celestia-da (implementation of go-da that embeds the whole celestia-node) and it seems that they are also taking a mandatory namespace parameter.

https://github.com/rollkit/celestia-da/blob/b5b95ff2fd15eb45a6aa399540f6ee9f22f8202d/cmd/celestia-da/server.go#L25-L26